lib/src/discovery_analysis.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Analyse a directory structure and find packages resolvers for each
+/// subdirectory.
+///
+/// The resolvers are generally the same that would be found by using
+/// the `discovery.dart` library on each subdirectory in turn,
+/// but more efficiently.
+library package_config.discovery_analysis;
+
+import "dart:collection" show HashMap;
+import "dart:io" show Directory, Platform;
+
+import "package:path/path.dart" as path;
+
+import "package_config.dart";
+import "discovery.dart";
+
+/// Associates a [PackageConfig] with a directory.
+///
+/// The package resolution applies to the directory and any subdirectory
+/// that doesn't have its own overriding child [PackageContext].
+abstract class PackageContext {
+  /// The directory which introduced the [packageCOnfig] resolver.
```

Comment:
Good catch (and *bad analyzer* for not checking doc comments :)

Reply:
> and _bad analyzer_ for not checking doc comments

Enable the [comment_references](https://dart-lang.github.io/linter/lints/comment_references.html) lint and it will check.